### PR TITLE
Feature/user task flow

### PR DESCRIPTION
**Pull request by Programmer-yar**
Run `python manage.py migrate` .

The user Task flow is implemented. These changes can be viewed at:
http://localhost:8000/polls/task-flow/
http://localhost:8000/polls/auth-flow/

The home page can be viewed at:
http://localhost:8000/
 
task_page
home
come_back
Screenshot from 2022-09-12 19-47-32

### CODE REVIEW

File: django_app/settings.py [hunk 1]
```diff
@@ -107,7 +107,11 @@
             'client_id': os.environ.get('GOOGLE_CLIENT_ID'),
             'secret': os.environ.get('GOOGLE_CLIENT_SECRET'),
             'key': ''
-        }
+        },
+        "SCOPE": [
```
Comment (turian) [inline django_app/settings.py:111]: What's SCOPE mean?

File: polls/views.py [hunk 2]
```diff
@@ -1,36 +1,70 @@
...
+            "should_rest": should_rest,
             "rest_time": rest_time
         }
-        return render(request, self.template_name, context)
\ No newline at end of file
+        return render(request, self.template_name, context)
+
+class TaskFlowView(LoginRequiredMixin, UserPassesTestMixin, View):
+    template_name = "polls/task_flow.html"
+
```
Comment (turian) [inline polls/views.py:30]: Are we currently checking if the user is frozen / banned? If not, just create an issue for that and do it later.

Comment (Programmer-yar) [reply polls/views.py:30]: Yes, in `test_func` under this class.

Comment (Programmer-yar) [reply polls/views.py:30]: if locked user tries to access `task-flow` page, they will see 403 (forbidden)

File: django_app/settings.py (hunk 1 above)
Comment (turian) [reply django_app/settings.py:111]: Can you add this as a comment?

File: polls/views.py [hunk 3]
```diff
@@ -1,36 +1,70 @@
...
+        Annotation.objects.create(
+            user=request.user,
+            task=Task.objects.get(pk=task_pk),
+            annotated_at=timezone.now(),
+            task_presentation=task_presentation,
+            annotations=annotation_choice
+        )
+        return redirect("task-flow")
+
+    def test_func(self):
```
Comment (turian) [inline polls/views.py:69]: This is a really generic name. Please rename to something better and more self-explanatory.

File: polls/views.py (hunk 2 above)
Comment (turian) [reply polls/views.py:30]: Can you post screenshots showing the steps from logging in to seeing the banned page?

Comment (Programmer-yar) [reply polls/views.py:30]: yes

File: polls/utils.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import random
+from django.utils import timezone
+
+def batch_selector():
+    random.seed()
```
[the remark this answers is not in the document]
Comment (Programmer-yar) [reply polls/utils.py:5]: Ok

File: polls/views.py (hunk 3 above)
Comment (Programmer-yar) [reply polls/views.py:69]: This is required by `UserPassesTestMixin` class, renaming will cause it malfunction

File: django_app/settings.py (hunk 1 above)
Comment (Programmer-yar) [reply django_app/settings.py:111]: Ok

File: polls/views.py (hunk 2 above)
Comment (Programmer-yar) [reply polls/views.py:30]: 403 and 404 pages can also be customized/styled

File: django_app/settings.py (hunk 1 above)
Comment (turian) [reply django_app/settings.py:112]: @Programmer-yar What is profile data? Can you point me to an example?

I have a feeling we only really want email. But show me what profile is with a screenshot

File: polls/views.py (hunk 2 above)
Comment (turian) [reply polls/views.py:30]: "
If `user.is_locked`, they are given a screen that says: "Thank you
for contributing." and can do no more work."

That's not a 403 or 404 page. Please create an issue to fix this later.

File: polls/views.py (hunk 3 above)
Comment (Programmer-yar) [reply polls/views.py:69]: Ok, I will implement it this way

File: polls/views.py (hunk 2 above)
Comment (Programmer-yar) [reply polls/views.py:30]: Ok

File: django_app/settings.py (hunk 1 above)
Comment (Programmer-yar) [reply django_app/settings.py:112]: Profile data include image url, Full name etc. But I agree with you we need email only for now

File: polls/views.py (hunk 3 above)
Comment (turian) [reply polls/views.py:69]: @Programmer-yar so link to the issue you created??

File: django_app/settings.py (hunk 1 above)
Comment (turian) [reply django_app/settings.py:112]: Make an issue and post a link to it here.

File: polls/views.py (hunk 2 above)
Comment (turian) [reply polls/views.py:30]: Please make an issue and link it here

File: django_app/settings.py (hunk 1 above)
Comment (Programmer-yar) [reply django_app/settings.py:112]: https://github.com/turian/audio-discrimination-crowdsource/issues/19

File: polls/views.py (hunk 2 above)
Comment (Programmer-yar) [reply polls/views.py:30]: https://github.com/turian/audio-discrimination-crowdsource/issues/20

File: polls/views.py (hunk 3 above)
Comment (Programmer-yar) [reply polls/views.py:69]: https://github.com/turian/audio-discrimination-crowdsource/issues/20

**Review by turian — APPROVED** [review]

**Pull request merged**